src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -1894,5 +2044,681 @@ struct S
+        }
+
+        [Fact]
+        public void Subsumption_12()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        _ = a switch
+        {
+            { Length: not 1 }  => 0,
+            [<0, ..] => 0,
+            [..[>= 0]] or [..null] => 1,
+            [_] => 2, // unreachable
+        };
+        _ = a switch // exhaustive
+        {
+            { Length: not 1 }  => 0,
+            [<0, ..] => 0,
+            [..[>= 0]] => 1,
+            [_] => 2,
+        };
+    }
+}" + TestSources.GetSubArray;
+            var comp = CreateCompilationWithIndexAndRange(src);
+            comp.VerifyEmitDiagnostics(
+                // (11,13): error CS8510: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.
+                //             [_] => 2, // unreachable
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "[_]").WithLocation(11, 13));
+        }
+
+        [Fact]
+        public void Subsumption_13()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        _ = a switch
+        {
+            [.., >0] => 1,
+            [<0, ..] => 2,
+            [0, ..] => 3,
+            { Length: not 1 } => 4,
+            [var unreachable] => 5,
+        };
+    }
+}" + TestSources.GetSubArray;
+            var comp = CreateCompilationWithIndexAndRange(src);
+            comp.VerifyEmitDiagnostics(
+                // (12,13): error CS8510: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.
+                //             [var unreachable] => 5,
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "[var unreachable]").WithLocation(12, 13));
+        }
+
+        [Fact]
+        public void Subsumption_14()
+        {
+            var src = @"
+class C
+{
+    void Test(int[] a)
+    {
+        _ = a switch
+        {
+            [.., >=0, >0] => 0,
+            [.., <0, >=0] => 0,
+            [<=0, <0, ..] => 1,
+            [>0, <=0, ..] => 1,
+            [0, 0] => 1,
+            { Length: not 2 }  => 2,
+            [var unreachable, var unreachable2] => 3,
+        };
+    }
+}";
+            var comp = CreateCompilationWithIndexAndRange(src);
+            comp.VerifyEmitDiagnostics(
+                // (14,13): error CS8510: The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.
+                //             [var unreachable, var unreachable2] => 3,
+                Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "[var unreachable, var unreachable2]").WithLocation(14, 13));
+        }
+
+        [Fact]
+        public void Subsumption_15()
+        {
+            // testing the scenario where we have multiple preconditions for indexers to relate.
```

Comment:
Then I don't understand this comment. Which indexers are we expecting to relate?